src/components/tabs/js/tabsController.js

```diff
@@ -169,6 +170,8 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
 
   function handleMaxTabWidth (newWidth, oldWidth) {
     if (newWidth !== oldWidth) {
+      var elements = getElements();
```

Comment:
You are correct, it does query the DOM, and we might be able to use a MutationObserver instead, but this function doesn't always get called from a MutationObserver. For example, it is called when the component is initialized.

If we have some hard data that suggests rewriting this to use the MutationObserver makes a drastic difference (verses querying the DOM on updates), then I'm all for it 😄